test/src/ui/vdom.spec.ts

```diff
@@ -0,0 +1,378 @@
+      expect(h.h2().tag).to.be('h2');
+      expect(h.h3().tag).to.be('h3');
+      expect(h.h4().tag).to.be('h4');
+      expect(h.h5().tag).to.be('h5');
+      expect(h.h6().tag).to.be('h6');
+      expect(h.header().tag).to.be('header');
+      expect(h.hr().tag).to.be('hr');
+      expect(h.i().tag).to.be('i');
+      expect(h.iframe().tag).to.be('iframe');
+      expect(h.img().tag).to.be('img');
+      expect(h.input().tag).to.be('input');
+      expect(h.ins().tag).to.be('ins');
+      expect(h.kbd().tag).to.be('kbd');
+      expect(h.label().tag).to.be('label');
+      expect(h.legend().tag).to.be('legend');
+      expect(h.li().tag).to.be('li');
+      expect(h.main().tag).to.be('main');
+      expect(h.map().tag).to.be('map');
+      expect(h.mark().tag).to.be('mark');
+      expect(h.meter().tag).to.be('meter');
+      expect(h.nav().tag).to.be('nav');
+      expect(h.noscript().tag).to.be('noscript');
+      expect(h.object().tag).to.be('object');
+      expect(h.ol().tag).to.be('ol');
+      expect(h.optgroup().tag).to.be('optgroup');
+      expect(h.option().tag).to.be('option');
+      expect(h.output().tag).to.be('output');
+      expect(h.p().tag).to.be('p');
+      expect(h.param().tag).to.be('param');
+      expect(h.pre().tag).to.be('pre');
+      expect(h.progress().tag).to.be('progress');
+      expect(h.q().tag).to.be('q');
+      expect(h.rp().tag).to.be('rp');
+      expect(h.rt().tag).to.be('rt');
+      expect(h.ruby().tag).to.be('ruby');
+      expect(h.s().tag).to.be('s');
+      expect(h.samp().tag).to.be('samp');
+      expect(h.section().tag).to.be('section');
+      expect(h.select().tag).to.be('select');
+      expect(h.small().tag).to.be('small');
+      expect(h.source().tag).to.be('source');
+      expect(h.span().tag).to.be('span');
+      expect(h.strong().tag).to.be('strong');
+      expect(h.sub().tag).to.be('sub');
+      expect(h.summary().tag).to.be('summary');
+      expect(h.sup().tag).to.be('sup');
+      expect(h.table().tag).to.be('table');
+      expect(h.tbody().tag).to.be('tbody');
+      expect(h.td().tag).to.be('td');
+      expect(h.textarea().tag).to.be('textarea');
+      expect(h.tfoot().tag).to.be('tfoot');
+      expect(h.th().tag).to.be('th');
+      expect(h.thead().tag).to.be('thead');
+      expect(h.time().tag).to.be('time');
+      expect(h.title().tag).to.be('title');
+      expect(h.tr().tag).to.be('tr');
+      expect(h.track().tag).to.be('track');
+      expect(h.u().tag).to.be('u');
+      expect(h.ul().tag).to.be('ul');
+      expect(h.var_().tag).to.be('var');
+      expect(h.video().tag).to.be('video');
+      expect(h.wbr().tag).to.be('wbr');
+    });
+
+  });
+
+  describe('realize()', () => {
+
+    it('should create a real DOM node from a virtual DOM node', () => {
+      let node = realize(h.div([h.a(), h.img()]));
+      expect(node.localName).to.be('div');
+      expect(node.firstChild.localName).to.be('a');
+      expect(node.lastChild.localName).to.be('img');
```

Comment:
`localName` should be `nodeName` (or equivalently `tagName`) which will be converted to uppercase:

https://developer.mozilla.org/en-US/docs/Web/API/Node/localName
https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeName